syntax = "proto3";
package georesearch;

service ProjectsService{
    // PROJECTS
  rpc CreateRecordProjects (CreateRequestProjects) returns (CreateResponseProjects);
  rpc ReadRecordProjects (ReadRequestProjects) returns (ReadResponseProjects);
  rpc UpdateRecordProjects (UpdateRequestProjects) returns (UpdateResponseProjects);
  rpc DeleteRecordProjects (DeleteRequestProjects) returns (DeleteResponseProjects);

}

message CreateRequestProjects {
  string name = 1;
  string description = 2;
  string status = 3;
}

message CreateResponseProjects {
  bool success = 1;
  string message = 2;
}


message ReadRequestProjects {
  int64 id = 1;
  string name = 2;
  string description = 3;
  string status = 4;
}

message ReadResponseProjects {
  int64 id = 1;
  string name = 2;
  string description = 3;
  string  timestamp = 4;
  string status = 5;
  bool success = 6;
  string message = 7;
}


message UpdateRequestProjects {
  int64 id = 1;
  string name = 2;
  string description = 3;
  string status = 4;
}

message UpdateResponseProjects {
  bool success = 1;
  string message = 2;
}


message DeleteRequestProjects {
  int64 id = 1;
  string name = 2;
  string description = 3;
  string status = 4;
}

message DeleteResponseProjects {  
  bool success = 1;
  string message = 2;
}